src/data-channel-wrapper.cpp

```diff
@@ -303,13 +294,22 @@ void DataChannelWrapper::onClosed(const Napi::CallbackInfo &info)
     mOnClosedCallback = std::make_unique<ThreadSafeCallback>(info[0].As<Napi::Function>());
 
     mDataChannelPtr->onClosed([&]()
-                              {
+                              {                                  
         if (mOnClosedCallback)
             mOnClosedCallback->call([this](Napi::Env env, std::vector<napi_value> &args) {
                 // Check the peer connection is not closed
                 if(instances.find(this) == instances.end())
                     throw ThreadSafeCallback::CancelException();
 
+                mOnOpenCallback.reset();
+                mOnClosedCallback.reset();
+                mOnErrorCallback.reset();
+                mOnBufferedAmountLowCallback.reset();
+                mOnMessageCallback.reset();
+
+                instances.erase(this);
+                mDataChannelPtr.reset();
```

Comment:
I think this works only if the user sets the `onClose` callback, otherwise the close event will be ignored. A solution might be moving the libdatachannel close callback setup to the constructor.

Reply:
That's right. I fixed it.

Reply:
Hmm, if `DataChannelWrapper::onClosed` is not called by the user, `mDataChannelPtr->onClosed` will never be set, so `cleanCbsAndEraseInstance()` will never be called. I think this can be solved by setting up `mDataChannelPtr->onClosed` in the constructor instead of in `DataChannelWrapper::onClosed`. 

I seems to me that the same mechanism would also allow to simplify `PeerConnectionWrapper`: Setting up `mRtcPeerConnPtr->onStateChange` in the constructor instead of in `PeerConnectionWrapper::onStateChange` would allow to always call the cleanup from the callback instead of relying on a boolean.